### ISSUE TITLE
Getting wrong errors on consecutive runs

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge#634]
On consecutive runs from the command line getting the following error
for the project [art 2.zip](https://github.com/getgauge/gauge/files/957307/art.2.zip)

command
```
mvn gauge:execute -DspecsDir=specs/one -DinParallel=true
```
Error
```
inblr-sswaroop:art sswaroop$ mvn gauge:execute -DspecsDir=specs/one -DinParallel=true
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building art 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- gauge-maven-plugin:1.1.0:execute (default-cli) @ art ---
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:20 Step implementation not found => 'Open the Gauge homepage "param"'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/GetStarted.cpt:7 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/GetStarted.cpt:3 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/GetStarted.cpt:4 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:28 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:29 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:30 Step implementation not found => 'a failure'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:34 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:35 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:39 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:40 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:44 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:45 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:49 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:50 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:54 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:55 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:59 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:60 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:64 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:65 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:69 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:70 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:74 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:75 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:79 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:80 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:84 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:85 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:89 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:90 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:94 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:95 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:99 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:100 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:104 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:105 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:109 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:110 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:114 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:115 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:119 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:120 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:124 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:125 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:129 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:130 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:134 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:135 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:139 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:140 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:144 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example1.spec:145 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:9 Step implementation not found => 'Open the Gauge homepage <param>'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:13 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:14 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:15 Step implementation not found => 'a failure'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:19 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:20 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:24 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:25 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:29 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:30 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:34 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:35 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:39 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:40 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:44 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:45 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:49 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:50 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:54 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:55 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:56 Step implementation not found => 'a failure'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:60 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:61 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:65 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:66 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:70 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:71 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:75 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:76 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:80 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:81 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:85 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:86 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:90 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:91 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:95 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:96 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:100 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:101 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:105 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:106 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:110 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:111 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:115 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:116 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:120 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:121 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:125 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:126 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:130 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:131 Step implementation not found => 'Ensure installation instructions are available'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:135 Step implementation not found => 'Go to Gauge Get Started Page'
[ValidationError] /Users/sswaroop/work/performance/gaugeTest/art/specs/one/example2.spec:136 Step implementation not found => 'Ensure installation instructions are available'
Executing in 8 parallel streams.
[runner: 2] # Getting Started with Gauge1
[runner: 2] 
Successfully generated html-report to => /Users/sswaroop/work/performance/gaugeTest/art/reports1/html-report/2017-04-26 11.05.58
Specifications:	1 executed	1 passed	0 failed	1 skipped
Scenarios:	0 executed	0 passed	0 failed	50 skipped
```

**Comment by kashishm** [comment]
@sguptatw Run `mvn test-compile gauge:execute -DspecsDir=specs/one -DinParallel=true` to compile test sources and run specs.

**Comment by sguptatw** [comment]
@kashishm : This happened soon after #633. The project was compiled. However after getting the connection closed issue #633. Consecutive run gave me this.

The problem is that these bugs are not occurring consistently. Infact after switching to IDE running the specs and switching back to command line is solving the problem

**Comment by kashishm** [comment]
@sguptatw This won't happen in IDE because IDE compiles the code first and then runs tests.

You can reproduce this by removing `target` directory and then run `mvn gauge:execute`. It will give you the above errors but running `mvn test-compile gauge:execute` will work.
